### PR TITLE
adds chungLuAlamEtAl generator + tests

### PR DESCRIPTION
**Pull request by PetholzA**
re-upload of #1166 from different fork

### CODE REVIEW

**Comment by fabratu** [comment]
Since the code from this PR uses #1178, a review will be added once the corresponding feature is merged.

**Review by fabratu — CHANGES_REQUESTED** [review]

File: networkit/cpp/generators/ChungLuGeneratorAlamEtAl.cpp [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        count degrees = std::min(degreeSequence[i], n - 1);
+        buckets[degrees]++;
+        sum_deg += degrees;
+    }
+    count maxGroupSize =
+        (count)(10
+                + n / std::sqrt(omp_get_max_threads()) / 2); // Make sure that the parallelization
+                                                             // step has enough groups to work with.
+    if (!parallel) {
+        maxGroupSize = 0xffffffffffffffffL;
```
[the remark this answers is not in the document]
Comment (PetholzA) [reply networkit/cpp/generators/ChungLuGeneratorAlamEtAl.cpp:36]: The part concerning the lambda function has been addressed, while i am not sure what exactly you mean by switching between `parallel` and `!parallel` in the outer level (and it does not seem necessary).

File: networkit/cpp/generators/ChungLuGeneratorAlamEtAl.cpp [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    return dist(rng);
+}
+
+template <typename F>
+void ChungLuGeneratorAlamEtAl::edgeSkipping(std::mt19937_64 &rng, F &&addEdge, index i, index j,
+                                            double p, index end) {
+    index x = 0 - 1;
+    double logP = 1.0 / std::log(1.0 - p);
+    while (true) {
+        double randVal = real(rng);
```
[the remark this answers is not in the document]
Comment (PetholzA) [reply networkit/cpp/generators/ChungLuGeneratorAlamEtAl.cpp:81]: While this is possible, as far as i can tell, Jannis used the preferred method described in https://networkit.github.io/dev-docs/cpp_api/function_namespaceAux_1_1Random_1ab0532015f7ed7d0eddffeb10405720f9.html for performance.

**Review by fabratu — CHANGES_REQUESTED** [review]

File: networkit/cpp/generators/ChungLuGeneratorAlamEtAl.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (PetholzA) [reply networkit/cpp/generators/ChungLuGeneratorAlamEtAl.cpp:36]: i resolved this slightly differently by adding a additional `if clause` and documenting the behaviour

File: include/networkit/generators/ChungLuGeneratorAlamEtAl.hpp [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+#include <networkit/generators/StaticGraphGenerator.hpp>
+
+namespace NetworKit {
+
+/**
+ * @ingroup generators
+ * Given an arbitrary degree sequence, the Chung-Lu generative model
+ * will produce a random graph with the same expected degree sequence.
+ *
+ * For reference of the concept of the original model see Chung, Lu: The average distances in random
```
Comment (fabratu) [inline include/networkit/generators/ChungLuGeneratorAlamEtAl.hpp:22]: `For reference, see the original model by Chung,...`

And use quotes to highlight the paper titles.

File: include/networkit/generators/ChungLuGeneratorAlamEtAl.hpp [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+ * graphs with given expected degrees and Chung, Lu: Connected Components in Random Graphs with
+ * Given Expected Degree Sequences. Aiello, Chung, Lu: A Random Graph Model for Massive Graphs
+ * describes a different generative model which is basically asymptotically equivalent but produces
+ * multi-graphs.
+ *
+ * This implementation follows a newer parallelizable algorithm by M. Alam, M. Khan, A. Vullikanti
```
Comment (fabratu) [inline include/networkit/generators/ChungLuGeneratorAlamEtAl.hpp:28]: Concerning author naming: Name two authors + et al. afterwards.

File: include/networkit/generators/ChungLuGeneratorAlamEtAl.hpp [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+ * and M. Marathe, "An Efficient and Scalable Algorithmic Method for Generating Large-Scale Random
+ * Graphs". https://ieeexplore.ieee.org/document/7877110 . The time complexity is O(m/p + delta +
```
Comment (fabratu) [inline include/networkit/generators/ChungLuGeneratorAlamEtAl.hpp:30]: Use markdown style for URLS: 

```
["An Efficient and Scalable Algorithmic Method for Generating Large-Scale Random Graphs"](https://ieeexplore.ieee.org/document/7877110)
```

Best also add URLs to the other references.

**Review by fabratu — CHANGES_REQUESTED** [review]

**Review by fabratu — APPROVED** [review]

**Pull request merged**